### PR TITLE
dialects: Stencil load op get method enhancement

### PR DESCRIPTION
**Pull request by mesham**
Support optional lower and upper bounds on the load op stencil .get method

### CODE REVIEW

File: xdsl/dialects/experimental/stencil.py [hunk 1]
```diff
@@ -451,13 +451,24 @@ class LoadOp(IRDLOperation):
...
 
+        attributes = {}
+        if lb is not None:
+            attributes["lb"] = lb
+        if ub is not None:
+            attributes["ub"] = ub
+
         return LoadOp.build(
             operands=[field],
+            attributes=attributes,
```
Comment (superlopuh) [inline xdsl/dialects/experimental/stencil.py:471]: nit:
```suggestion
            attributes={
                "lb": lb,
                "ub": ub,
            },
```
We now handle optionals, so this should work

**Review by superlopuh — APPROVED** [review]

**Review by AntonLydike — APPROVED** [review]
LGTM modulo sashas comment

File: xdsl/dialects/experimental/stencil.py [hunk 2]
```diff
@@ -468,7 +468,10 @@ def get(
 
         return LoadOp.build(
             operands=[field],
-            attributes=attributes,
+            attributes={
```
Comment (superlopuh) [inline xdsl/dialects/experimental/stencil.py:471]: Can you please remove the unused attributes dict?

Comment (mesham) [reply xdsl/dialects/experimental/stencil.py:394]: Thanks, done :)

File: xdsl/dialects/experimental/stencil.py (hunk 1 above)
Comment (mesham) [reply xdsl/dialects/experimental/stencil.py]: Ah, nice - thanks :)

**Pull request merged**